game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProBattleUtils.java

```diff
@@ -391,8 +395,26 @@ public static boolean territoryHasLocalNavalSuperiority(
             + myUnits.size()
             + ", enemySize="
             + enemyUnitsInSeaTerritories.size());
+    boolean negativeAttackTuv = false;
+    if (attackStrengthDifference > 50) {
+      // To really have naval attack superiority, ensure there's also a positive attack TUV for
+      // all nearby territories. Otherwise, prioritizeAttackOptions() will exclude such attacks,
+      // thus coming short of actually having naval superiority.
+      for (final Territory nearby : nearbyEnemySeaTerritories) {
+        final List<Unit> enemySeaUnits = nearby.getUnitCollection().getMatches(enemyNonLandUnit);
+        if (enemySeaUnits.isEmpty()) {
+          continue;
+        }
+        ProBattleResult result =
+            calc.estimateAttackBattleResults(proData, t, myUnits, enemySeaUnits, List.of());
```

Comment:
Not sure how often `territoryHasLocalNavalSuperiority` is called and what the size of `nearbyEnemySeaTerritories` can be but you'll want to make sure this doesn't have a negative performance impact as I believe this method can call the battle calc.

Reply:
Good point. I've now changed it to just to a battle calc against the strongest defensive enemy fleet only, which should be sufficient.